nab/src/org/labkey/nab/query/NabControlAggregatesTable.java

```diff
@@ -0,0 +1,99 @@
+package org.labkey.nab.query;
+
+import org.jetbrains.annotations.NotNull;
+import org.labkey.api.assay.AssayProvider;
+import org.labkey.api.assay.AssayService;
+import org.labkey.api.data.BaseColumnInfo;
+import org.labkey.api.data.ColumnInfo;
+import org.labkey.api.data.ContainerFilter;
+import org.labkey.api.data.ContainerFilterable;
+import org.labkey.api.data.JdbcType;
+import org.labkey.api.data.SQLFragment;
+import org.labkey.api.data.TableInfo;
+import org.labkey.api.data.VirtualTable;
+import org.labkey.api.exp.api.ExpProtocol;
+import org.labkey.api.query.LookupForeignKey;
+import org.labkey.api.util.StringExpression;
+
+import static org.labkey.api.assay.dilution.DilutionManager.WELL_DATA_TABLE_NAME;
+
+public class NabControlAggregatesTable extends VirtualTable<NabProtocolSchema> implements ContainerFilterable
+{
+    private final String _controlName;
+
+    public NabControlAggregatesTable(String tableName, String controlName, NabProtocolSchema schema, ContainerFilter cf, ExpProtocol protocol, AssayProvider provider)
+    {
+        super(schema.getDbSchema(), tableName, schema, cf);
+        _controlName = controlName;
+        setDescription("Contains one row per run for the aggregate values of the control well data values.");
+
+        var runCol = new BaseColumnInfo("RunId", this, JdbcType.INTEGER);
+        runCol.setFk(new LookupForeignKey(getContainerFilter(), "RowID", null)
```

Comment:
I always forget about this, but consider in the future using the variant of setFk which takes a `Builder<ForeignKey>` argument. Matt introduced this with the mutable ColumnInfo changes and its pretty nifty and can yield a more concise syntax for FK creation.

Reply:
Ah, I see you already know about this patter 👍

Reply:
Good catch. I was just copying that run column FK from another NAb table, but have switched it to `runCol.setFk(QueryForeignKey.from(schema, cf).to("Runs", "RowID", null));`